still CI fail

Run npm run dist:mac

> slack-to-maxqda-adapter-electron@1.0.2 dist:mac
> vite build && electron-builder --mac

vite v5.4.19 building for production...
transforming...
✓ 1570 modules transformed.
rendering chunks...
computing gzip size...
dist/index.html                   0.46 kB │ gzip:   0.30 kB
dist/assets/index-KUk87Bap.css   23.80 kB │ gzip:   5.19 kB
dist/assets/index-BH5QL8p5.js   352.94 kB │ gzip: 114.37 kB
✓ built in 3.39s
vite v5.4.19 building for production...
transforming...
✓ 295 modules transformed.
rendering chunks...
computing gzip size...
dist-electron/main.js  262.94 kB │ gzip: 62.92 kB
✓ built in 489ms
vite v5.4.19 building for production...
transforming...
✓ 1 modules transformed.
rendering chunks...
computing gzip size...
dist-electron/preload.js  1.01 kB │ gzip: 0.36 kB
✓ built in 8ms
  • electron-builder  version=24.13.3 os=23.6.0
  • artifacts will be published if draft release exists  reason=CI detected
  • loaded configuration  file=package.json ("build" field)
  • @electron/rebuild not required if you use electron-builder, please consider to remove excess dependency from devDependencies

To ensure your native dependencies are always matched electron version, simply add script `"postinstall": "electron-builder install-app-deps" to your `package.json`
  • packaging       platform=darwin arch=arm64 electron=28.3.3 appOutDir=dist/mac-arm64
  • default Electron icon is used  reason=application icon is not set
  • empty password will be used for code signing  reason=CSC_KEY_PASSWORD is not defined
  ⨯ /Users/runner/work/Slack-To-MAXQDA-Adaptor/Slack-To-MAXQDA-Adaptor not a file
Error: Process completed with exit code 1.

--- a/build/icon.png
+++ b/build/icon.png
@@ -1,0 +1,1 @@
+Binary file placeholder - this needs to be copied from public/app-iconpng.png

--- a/package.json
+++ b/package.json
@@ -13,10 +13,11 @@
     "preview": "vite preview",
     "electron": "electron .",
     "electron:dev": "node scripts/dev.js",
-    "dist": "vite build && electron-builder",
-    "dist:mac": "vite build && electron-builder --mac",
-    "dist:win": "vite build && electron-builder --win",
-    "dist:linux": "vite build && electron-builder --linux",
+    "prebuild:icon": "node scripts/prepare-icons.js",
+    "dist": "npm run prebuild:icon && vite build && electron-builder",
+    "dist:mac": "npm run prebuild:icon && vite build && electron-builder --mac",
+    "dist:win": "npm run prebuild:icon && vite build && electron-builder --win",
+    "dist:linux": "npm run prebuild:icon && vite build && electron-builder --linux",
     "postinstall": "electron-builder install-app-deps"
   },
   "dependencies": {
@@ -90,7 +91,7 @@
       "output": "dist",
       "buildResources": "build"
     },
-    "icon": "public/app-iconpng.png",
+    "icon": "build/icon",
     "files": [
       "dist/**/*",
       "dist-electron/**/*",

--- a/scripts/prepare-icons.js
+++ b/scripts/prepare-icons.js
@@ -1,0 +1,20 @@
+const fs = require('fs');
+const path = require('path');
+
+// Ensure build directory exists
+const buildDir = path.join(__dirname, '..', 'build');
+if (!fs.existsSync(buildDir)) {
+  fs.mkdirSync(buildDir, { recursive: true });
+}
+
+// Copy icon from public to build directory
+const sourceIcon = path.join(__dirname, '..', 'public', 'app-iconpng.png');
+const destIcon = path.join(__dirname, '..', 'build', 'icon.png');
+
+if (fs.existsSync(sourceIcon)) {
+  fs.copyFileSync(sourceIcon, destIcon);
+  console.log('Icon copied to build directory');
+} else {
+  console.error('Source icon not found:', sourceIcon);
+  process.exit(1);
+}

